admin/assessmentExport.php

```diff
@@ -18,41 +18,60 @@
 {
   $cid = $_GET['cid'];
   
-  $assessment_array= array();  
-
-  $query = "SELECT imas_assessments.id,imas_assessments.name FROM `imas_assessments` 
-  join imas_courses 
-  ON imas_assessments.courseid = imas_courses.id
-  WHERE imas_courses.id = ".$cid .";" ;
+  $modules_array= array(); 
+  $assessment_names_array= array(); 
+  $module= loop1(1);
   
-  $result = mysql_query($query) or die("Query failed : " . mysql_error());
+  foreach ($module as $block) {
```

Comment:
Be careful here.  At any stage of the itemorder array, an element can either be an imas_items.id, or an array (indicating it is a block/module with a $block['items'] element. It looks like you're assuming every element of the array will be a block, and aren't accounting for blocks inside blocks.  You're also assuming that every item is an assessment, which is not true.  Oh, and you're using the imas_items.id as the assessment ID, which is not true.

Two options:
1) Just pull all assessments from imas_assessments using courseid as the filter, and don't worry about the itemorder at all, or

2) Recurse through itemorder, and pull from imas_items.  Maybe something like (untested):
````
$query = "SELECT id,typeid FROM imas_items WHERE courseid='$cid' AND itemtype='Assessment";
$result = mysql_query($query);
while ($row = mysql_fetch_assoc($result)) {
  $assessitems[$row['id']] = $row['typeid']; //this stores imas_items->assessmentid relations
}
function recurseitemorder($items,&$assessments) {
  foreach ($items as $item) {
    if (is_array($item)) {  //is a block - process contents
       recurseitemorder($item['item'], $assessments);
    } else if (isset($assessitems[$item])) { //if this item is an assessment, add to array
      $assessments[] = $assessitems[$item];
   }
}
$assessments = array();
recurseitemorder($itemorder, $assessments);
````